src/main/java/frc/robot/vision/rawdata/RawVisionData.java

```diff
@@ -0,0 +1,24 @@
+package frc.robot.vision.rawdata;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import frc.robot.poseestimator.observations.VisionRobotPoseObservation;
+
+public class RawVisionData implements VisionRobotPoseObservation {
```

Comment:
maybe make RawVisionData interface, so it solve u all the funny T extends

Reply:
If anything, then abstract class. But some cameras could return RawVisionData (after all, it's enough to track one target. The problem is mostly when you track two targets, like both the robot and the apriltag).

Reply:
But why not interface? It’s solving the extends problem in all of the code…
U can also create another interface and implement it if u really care about the getters duplicate code

Reply:
ok right

Reply:
like T extends, the same solution would work here: use instanceOf
Butt generally needing to dive into instance of interfaces or abstraction is not good